src/stk/image/gpu_volume.cpp

```diff
@@ -118,6 +118,65 @@ namespace
 
         return vol;
     }
+
+    cudaMemcpy3DParms make_download_params(const GpuVolume& src, const Volume& dst)
+    {
+        cudaMemcpy3DParms params = {0};
+
+        // We also assume both volumes have same dimensions
+        ASSERT(dst.size() == src.size());
+        ASSERT(dst.voxel_type() == src.voxel_type());
+
+        dim3 size = dst.size();
+
+        params.kind = cudaMemcpyDeviceToHost;
+        params.dstPtr = make_cudaPitchedPtr(const_cast<void*>(dst.ptr()),
+            dst.strides()[1], dst.strides()[1]/dst.strides()[0], dst.strides()[2]/dst.strides()[1]);
```

Comment:
Kinda backwards, but essentially extracts the dimensions of the backing memory using the strides.